tools/afl-fuzzer/CMakeLists.txt

```diff
@@ -0,0 +1,66 @@
+# SPDX-License-Identifier: BSD-3-Clause
+
+cmake_minimum_required(VERSION 3.10)
+
+project(SOF_AFL_FUZZER C)
+
+if("${CMAKE_CURRENT_SOURCE_DIR}" STREQUAL "${CMAKE_CURRENT_BINARY_DIR}")
+	message(FATAL_ERROR
+		" In-source builds are not supported.\n"
+		" Please remove CMakeCache.txt and the CMakeFiles directory.\n"
+		" Then specify a build directory. Example: cmake -Bbuild ..."
+	)
+endif()
+
+include(ExternalProject)
+
+set(parser_src_dir "${PROJECT_SOURCE_DIR}/../tplg_parser")
+set(parser_install_dir "${PROJECT_BINARY_DIR}/sof_parser/install")
+
+ExternalProject_Add(sof_parser_ep
+	SOURCE_DIR "${parser_src_dir}"
+	PREFIX "${PROJECT_BINARY_DIR}/sof_parser"
+	BINARY_DIR "${PROJECT_BINARY_DIR}/sof_parser/build"
+	CMAKE_ARGS -DCMAKE_INSTALL_PREFIX=${parser_install_dir}
+		-DCMAKE_VERBOSE_MAKEFILE=${CMAKE_VERBOSE_MAKEFILE}
+	BUILD_COMMAND ${CMAKE_COMMAND}
+	BUILD_ALWAYS 1
+	BUILD_BYPRODUCTS "${parser_install_dir}/lib/libsof_tplg_parser.so"
+)
+
+add_library(sof_parser SHARED IMPORTED)
+set_target_properties(sof_parser PROPERTIES IMPORTED_LOCATION "${parser_install_dir}/lib/libsof_tplg_parser.so")
+add_dependencies(sof_parser sof_parser_ep)
+
+add_executable(sof-afl-fuzzer
```

Comment:
@ymdatta I am a bit confused. If we're reusing most of the includes and the qemu-bridge and platform code, cant we simply create a new main.c for the afl-fuzzer and just add that executable to the cmakelists.txt in tools/fuzzer? It seems a bit of an overkill to duplicate everything there, no?

Reply:
Yes. It's an over kill. I updated to create a new executable for sof-afl-fuzzer.